[test] Avoid writing to a potentially write-protected dir

karka228 opened the pull request:
The test length-errors.hlsl don't check the output written to the current directory. The current directory may be write protected e.g. in a sandboxed environment.

This patch simply remove the -emit-llvm option as this testcase don't care about the outputed llvm IR.

karka228 committed:
[test] Avoid writing to a potentially write-protected dir

The test length-errors.hlsl don't check the output written to the current
directory. The current directory may be write protected e.g. in a sandboxed
environment.

This patch remove the -emit-llvm option as these testcases don't care about the
outputed llvm IR.

Co-authored-by: Chris B <cbieneman@microsoft.com>

llvm-beanz commented on clang/test/SemaHLSL/BuiltIns/length-errors.hlsl:
```diff
@@ -1,4 +1,4 @@
-// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -fnative-half-type -emit-llvm -disable-llvm-passes -verify -verify-ignore-unexpected
+// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -fnative-half-type -emit-llvm-only -disable-llvm-passes -verify -verify-ignore-unexpected
```
This probably shouldn't have the -emit-llvm flag at all since it isn't actually checking that output.

llvm-beanz replied:
```suggestion
// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -fnative-half-type -disable-llvm-passes -verify -verify-ignore-unexpected

```

llvm-beanz reviewed (commented):
Looks like neither of these tests should be passing `-emit-llvm` at all.

FreddyLeaf commented on clang/test/CodeGen/X86/avx10_2_512minmax-error.c:
```diff
@@ -1,7 +1,7 @@
 // RUN: %clang_cc1 %s -flax-vector-conversions=none -ffreestanding -triple=x86_64 -target-feature +avx10.2-512 \
-// RUN: -Wno-invalid-feature-combination -emit-llvm -verify
+// RUN: -Wno-invalid-feature-combination -emit-llvm-only -verify
```
I just realized `-emit-llvm is not required here. You can directly removed it. Sorry for mistake.

llvm-beanz replied:
```suggestion
// RUN: -Wno-invalid-feature-combination -verify
```

llvm-beanz commented on clang/test/CodeGen/X86/avx10_2_512minmax-error.c:
```diff
@@ -1,7 +1,7 @@
+// RUN: -Wno-invalid-feature-combination -emit-llvm-only -verify
 // RUN: %clang_cc1 %s -flax-vector-conversions=none -ffreestanding -triple=i386 -target-feature +avx10.2-512 \
-// RUN: -Wno-invalid-feature-combination -emit-llvm -verify
+// RUN: -Wno-invalid-feature-combination -emit-llvm-only -verify
```
```suggestion
// RUN: -Wno-invalid-feature-combination -verify
```

karka228 commented:
When rebasing the patch I removed the fix in avx10_2_512minmax-error.c as it was already fixed in:

```
commit 1b9d7dd9eb31974077b64d66404193bd7c4ad65e
Author: Augie Fackler <augie@google.com>
Date:   Mon Aug 5 17:08:06 2024 -0400

    tests: fix clang flags in avx10_2_512minmax-error test
    
    Introduced in 3d5cc7e1e632b74119af13824dabc346bd248c93, caught by a
    sandboxed test run.
```
